assets/js/base/components/payment-methods/express-payment-methods.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * External dependencies
+ */
+import {
+	useCheckoutData,
+	usePaymentEvents,
+	useExpressPaymentMethods,
+} from '@woocommerce/base-hooks';
+import { cloneElement } from '@wordpress/element';
+
+const ExpressPaymentMethods = () => {
+	const [ checkoutData ] = useCheckoutData();
+	const { dispatch, select } = usePaymentEvents();
+	// not implementing isInitialized here because it's utilized further
+	// up in the tree for express payment methods. We won't even get here if
+	// there's no payment methods after initialization.
+	const { paymentMethods } = useExpressPaymentMethods();
+	const paymentMethodSlugs = Object.keys( paymentMethods );
+	const content =
+		paymentMethodSlugs.length > 0 ? (
```

Comment:
> I think there is no need to check paymentMethodSlugs's length before calling paymentMethodSlugs.map(): .map() will do nothing if it's an empty array.

Right, but I'm also conditionally outputting something different if the array is empty.  The logic could be written differently to get the same result but I think this is a case of potatoe/potahto :)

Reply:
> Besides that, can't you directly iterate on paymentMethods instead of extracting the slugs, iterating on the slugs, and then getting the payment method based on the slug?

Hmm probably, again though, I'm using the "empty" check to conditionally output the "No Registered Payments" list item.  Checking if an object is empty isn't as straightforward.  Another case of the logic can be done differently but what does it gain to do so? (serious question... I may be missing something obvious).

Reply:
> Right, but I'm also conditionally outputting something different if the array is empty.

Oh right, I missed that. I think the current implementation makes sense so just ignore my previous comment. :slightly_smiling_face: